components/codec/src/number.rs

```diff
@@ -708,6 +714,26 @@ macro_rules! write {
 }
 
 pub trait BufferNumberEncoder: BufferWriter {
+    #[inline]
+    fn write_u8(&mut self, v: u8) -> Result<()> {
+        write!(self, v, 1, encode_u8)
+    }
+
+    #[inline]
+    fn write_all(&mut self, values: &[u8]) -> Result<()> {
+        let buf = unsafe { self.bytes_mut(values.len()) };
+        if buf.len() < values.len() {
+            return Err(Error::BufferTooSmall);
+        }
+        for i in 0..values.len() {
```

Comment:
`copy_from_slice` is more efficient.

Reply:
`copy_from_slice` needs src and dest must have the exactly same length, here the length of `buf` maybe greater than `value`'s

Reply:
How about something like:

```rust
buf[0..values.len()].copy_from_slice(values)
```